hudi-common/src/main/java/org/apache/hudi/common/util/collection/ExternalSpillableMap.java

```diff
@@ -108,7 +116,7 @@ public ExternalSpillableMap(Long maxInMemorySizeInBytes, String baseFilePath, Si
                 break;
               case BITCASK:
               default:
-                diskBasedMap = new BitCaskDiskMap<>(baseFilePath);
+                diskBasedMap = isCompressionEnabled ? (new BitCaskDiskMap<>(baseFilePath, true)) : (new BitCaskDiskMap<>(baseFilePath, false));
```

Comment:
new BitCaskDiskMap<>(baseFilePath, isCompressionEnabled)

Reply:
lol this was stupid :)